src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,244 @@
+
+        Returns:
+            None
+        '''
+        keyname = Gdk.keyval_name(event.keyval)
+        if keyname == 'Escape':
+            self.destroy()
+
+    def __realize_cb(self, widget):
+        '''
+        realize signal handler
+        Sets window as a transient window for the window specified by parent.
+
+        Args:
+            widget (Gtk.Widget): the widget(window) which received the signal
+
+        Returns:
+            None
+        '''
+        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)
+        window = self.get_window()
+        window.set_accept_focus(True)
+
+        if self._parent_window_xid is not None:
+            display = Gdk.Display.get_default()
+            parent = GdkX11.X11Window.foreign_new_for_display(
+                display, self._parent_window_xid)
+            window.set_transient_for(parent)
+            shell.get_model().push_modal()
+
+    def __hide_cb(self, widget):
+        '''
+        hide signal handler
+
+        Args:
+            widget (Gtk.Widget): the widget(window) which received the signal
+
+        Returns:
+            None
+        '''
+        shell.get_model().pop_modal()
+
+    def add_view(self, widget, expand=True, fill=True, padding=0):
+        '''
+        Adds child to the vbox.
+
+        Args:
+            widget (Gtk.Widget): widget to be added
+
+            expand (bool): True if child is to be given extra space allocated
+                to vbox.
+
+            fill (bool): True if space given to child by the expand option is
+                actually allocated to child, rather than just padding it.
+
+            padding (int): extra space in pixels to put between child and its
+                neighbors, over and above the global amount specified
+                by spacing in vbox.
+
+        Returns:
+            None
+        '''
+        self._vbox.pack_start(widget, True, fill, padding)
+
+
+class TitleBox(ToolbarBox):
+    '''
+    Title box at the top of the pop-up window.
+    Title and close button are added to the box and as needed more widgets
+    can be added using self.add_widget method.
+    '''
+
+    def __init__(self):
+        ToolbarBox.__init__(self)
+
+        self.close_button = ToolButton(icon_name='dialog-cancel')
+        self.close_button.set_tooltip(_('Close'))
+        self.toolbar.insert(self.close_button, -1)
+        self.close_button.show()
+        self._label = Gtk.Label()
+        self._label.set_alignment(0, 0.5)
+        self.add_widget(self._label, expand=True, position=0)
+
+    def set_title(self, title):
+        '''
+        setter function for 'title' property.
+        Args:
+           title (str): title for the pop-up window
+        '''
+        self._label.set_markup('<b>%s</b>' % title)
+        self._label.show()
+
+    title = GObject.property(type=str, setter=set_title)
+
+    def add_widget(self, widget, expand=False, position=-1):
+        '''
+        Adds widget to the title-box.
```

Comment:
Document that this func wraps it in a toolitem, and that you can just insert buttons via `x.props.toolbar.insert(...)`